src/algorithms/standard/constantq.cpp

```diff
@@ -70,76 +72,87 @@ void ConstantQ::configure() {
   _numberBins = parameter("numberBins").toInt();
   _binsPerOctave = parameter("binsPerOctave").toInt();
   _threshold = parameter("threshold").toDouble();
+  _scale = parameter("scale").toDouble();
+
+  _windowing->configure("type", parameter("windowType").toString());
 
   // Constant Q factor (resolution of filter windows, larger values correspond
   // to longer windows.
-  // TODO make filterScale configurable (0,+inf) (default = 1)?
-  const Real filterScale = 1.;
-  _Q = filterScale / (pow(2, (1/(double)_binsPerOctave))-1);
+  _Q = _scale / (pow(2, (1/(double)_binsPerOctave))-1);
   
   // The largest window size we'll require. We center-pad filters to the next
   // power of two of the maximum filter length.
-  _windowSize = nextPowerTwo((int) ceil(_Q *_sampleRate/_minFrequency));
+  _windowSize = nextPowerTwo((int)ceil(_Q * _sampleRate / _minFrequency));
+
+  _fft->configure("size", _windowSize);
+  _fft->output("fft").set(_fftData);
 
   // Work only with a non-negative part of FFT as an input
-  _inputFFTSize = _windowSize/2+1;
+  _inputFFTSize = _windowSize / 2 + 1;
+
+  _sparseKernel = {};
 
-  struct SparseKernel &sk = _sparseKernel;
-  sk.i.reserve(_windowSize);
```

Comment:
What's the rationalle  not using `reserve()`? They are here to make interative ``push_back()`` calls faster.

Reply:
I just though that the code will be more clear without these references and forgot to keep the `reserves` when removing them. I'll put them back.